src/modules/form/components/DynamicFormFields.tsx

```diff
@@ -0,0 +1,242 @@
+          if (!enabledDependencyMap[changedLinkId]) return;
+
+          enabledDependencyMap[changedLinkId].forEach((dependentLinkId) => {
+            const enabled = shouldEnableItem(
+              itemMap[dependentLinkId],
+              localValues,
+              itemMap
+            );
+            if (enabled && newList.includes(dependentLinkId)) {
+              pull(newList, dependentLinkId);
+            } else if (!enabled && !newList.includes(dependentLinkId)) {
+              newList.push(dependentLinkId);
+            }
+          });
+        });
+
+        return newList;
+      });
+    },
+    [itemMap, enabledDependencyMap, setDisabledLinkIds]
+  );
+
+  const itemChanged = useCallback(
+    (linkId: string, value: any) => {
+      if (itemsChangedProp) itemsChangedProp(linkId, value);
+      setValues((currentValues) => {
+        const newValues = { ...currentValues };
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        newValues[linkId] = value;
+
+        // These must fire on the next render cycle otherwise it's a react lifecycle error
+        setTimeout(() => {
+          // Updates dependent autofill questions (modifies newValues in-place)
+          updateAutofillValues([linkId], newValues);
+          // Update list of disabled linkIds based on new values
+          updateDisabledLinkIds([linkId], newValues);
+        });
+
+        return newValues;
+      });
+    },
+    [updateAutofillValues, updateDisabledLinkIds, setValues, itemsChangedProp]
+  );
+
+  const severalItemsChanged = useCallback(
+    (values: Record<string, any>) => {
+      if (severalItemsChangedProp) severalItemsChangedProp(values);
+      setValues((currentValues) => {
+        const newValues = { ...currentValues, ...values };
+        // Updates dependent autofill questions (modifies newValues in-place)
+        updateAutofillValues(Object.keys(values), newValues);
+        // Update list of disabled linkIds based on new values
+        updateDisabledLinkIds(Object.keys(values), newValues);
+        // console.debug('DynamicForm', newValues);
+        return newValues;
+      });
+    },
+    [
+      updateDisabledLinkIds,
+      updateAutofillValues,
+      setValues,
+      severalItemsChangedProp,
+    ]
+  );
+
+  // Get errors for a particular field
+  const getFieldErrors = useCallback(
+    (item: FormItem) => {
+      if (!errors) return undefined;
+      if (!item.fieldName) return undefined;
+      const attribute = item.fieldName;
+      return errors.filter(
+        (e) => e.attribute === attribute && e.type !== CONFIRM_ERROR_TYPE
+      );
+    },
+    [errors]
+  );
+
+  // Recursively render an item
+  const renderItem = (
+    item: FormItem,
+    nestingLevel: number,
+    props?: OverrideableDynamicFieldProps,
+    renderFn?: (children: ReactNode) => ReactNode
+  ) => {
+    const isDisabled = !isEnabled(item, disabledLinkIds);
+    if (!isShown(item, disabledLinkIds)) return null;
```

Comment:
This is running `isEnabled` twice which is unnecessary and potentially expenseive.

I think it's clearer written as it was before:

```suggestion
    const isDisabled = !isEnabled(item, disabledLinkIds);
    if (isDisabled && item.disabledDisplay !== DisabledDisplay.Protected) return null;
```